ui/src/app/auth/login/login.component.ts

```diff
@@ -7,9 +8,25 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  loginForm : FormGroup ;
+  formSubmitted : boolean = false;
+
+  constructor(private fb:FormBuilder) { }
 
   ngOnInit(): void {
+    this.loginForm = this.fb.group({
+      'email': ['', Validators.compose([Validators.required, Validators.pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)])],
+      'password': ['', Validators.compose([Validators.required])]
+  });
+  }
+
+  onSubmit() {
+      if (this.loginForm.valid) {
+        this.formSubmitted = true;
```

Comment:
by this way the user unable to see the errors, this should be in the first

Reply:
Why did the user not see the errors?

Reply:
Done